File: app/src/main/java/com/solution/it/newsoft/dagger/ActivityBindingModules.java
package com.solution.it.newsoft.dagger;

import com.solution.it.newsoft.ui.ListingActivity;
import com.solution.it.newsoft.ui.LoginActivity;
import com.solution.it.newsoft.ui.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
abstract class ActivityBindingModules {

    @ContributesAndroidInjector
    abstract MainActivity contributeMainActivity();

    @ContributesAndroidInjector
    abstract LoginActivity contributeLoginActivity();

    @ContributesAndroidInjector
    abstract ListingActivity contributeListingActivity();
}

File: app/src/main/java/com/solution/it/newsoft/dagger/ActivityBindingModules.kt
package com.solution.it.newsoft.dagger

import com.solution.it.newsoft.ui.ListingActivity
import com.solution.it.newsoft.ui.LoginActivity
import com.solution.it.newsoft.ui.MainActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
internal abstract class ActivityBindingModules {

    @ContributesAndroidInjector
    internal abstract fun contributeMainActivity(): MainActivity

    @ContributesAndroidInjector
    internal abstract fun contributeLoginActivity(): LoginActivity

    @ContributesAndroidInjector
    internal abstract fun contributeListingActivity(): ListingActivity
}
